lib/elixir/lib/code.ex

```diff
@@ -1045,6 +1045,66 @@ defmodule Code do
   When given a path to a `.beam` file, it will load the docs directly from that
   file.
 
+  It returns the term stored in the documentation chunk in the format defined by
+  [EEP 48](http://erlang.org/eep/eeps/eep-0048.html) or `{:error, reason}` if
+  the chunk is not available.
+
+  ## Examples
+
+      # Module documentation of an existing module
+      iex> {:docs_v1, _, :elixir, _, %{"en" => module_doc}, _, _} = Code.fetch_docs(Atom)
+      iex> module_doc |> String.split("\n") |> Enum.at(0)
+      "Convenience functions for working with atoms."
+
+      # A module that doesn't exist
+      iex> Code.fetch_docs(ModuleNotGood)
+      {:error, :module_not_found}
+
+  """
+  @spec fetch_docs(module) ::
+          {:docs_v1, anno, beam_language, format, module_doc :: doc, metadata,
+           docs :: [{{kind, name, arity}, anno, signature, doc, metadata}]}
+          | {:error, reason :: term}
+        when anno: :erl_anno.anno(),
+             beam_language: atom,
+             format: binary,
+             doc: %{binary => binary} | :none | :hidden,
+             kind: atom,
+             name: atom,
+             signature: [binary],
+             metadata: map
+  @since "1.7.0"
+  def fetch_docs(module) when is_atom(module) do
+    case :code.get_object_code(module) do
```

Comment:
@josevalim I'm not sure if the current implementation addresses the case when the chunk is not in the beam file, but in a separate "doc/chunks/example.chunk" file. Would this be resolved in by `:beam_lib.chunks/2` or do we need to handle that case ourselves? This is where a good example case from an Erlang module would help.

Reply:
We would need to handle it ourselves but let's skip it for now as no project seems to be using the doc/chunks bit.